common/dds-model.cpp

```diff
@@ -0,0 +1,349 @@
+    ImGui::PushStyleColor( ImGuiCol_Text, light_grey );
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowPadding, ImVec2( 5, 5 ) );
+    ImGui::PushStyleVar( ImGuiStyleVar_WindowRounding, 1 );
+    ImGui::PushStyleColor( ImGuiCol_Button, button_color );
+    ImGui::PushStyleColor( ImGuiCol_ButtonHovered, button_color + 0.1f );
+    ImGui::PushStyleColor( ImGuiCol_ButtonActive, button_color + 0.1f );
+
+
+    if( ImGui::BeginPopupModal( window_name, nullptr, flags ) )
+    {
+        if( error_message != "" )
+            ImGui::CloseCurrentPopup();
+
+        // Title
+        const char * title_message = window_name;
+        ImVec2 title_size = ImGui::CalcTextSize( title_message );
+        float title_x = ( w - title_size.x - 10 ) / 2.0f;
+        ImGui::SetCursorPos( { title_x, 10.0f } );
+        ImGui::PushFont( window.get_large_font() );
+        ImGui::PushStyleColor( ImGuiCol_Text, white );
+        ImGui::Text( "%s", title_message );
+        ImGui::PopStyleColor();
+        ImGui::PopFont();
+        ImGui::Separator();
+        ImGui::SetCursorPosY( ImGui::GetCursorPosY() + 15 );
+
+        // Main Scrollable Section
+        ImGui::BeginChild( "MainContent", ImVec2( w - 10, h - 100 ), true );
+        ImGui::PushItemWidth( 150.0f );
+
+        // Connection Priority Section
+        priority connection_priority = classifyPriority( _changed_config.link.priority );
+        if( ImGui::CollapsingHeader( "Connection Priority" ) )
+        {
+            ImGui::Text( "Select connection priority:" );
+            ImGui::RadioButton( "Ethernet First", reinterpret_cast< int * >( &connection_priority ), 0 );
+            if( static_cast< int >( connection_priority ) == 0 )
+            {
+                ImGui::SameLine();
+                ImGui::SetCursorPosX( ImGui::GetCursorPosX() + 50 );
+                ImGui::Text( "Link Timeout (seconds)" );
+                ImGui::SameLine();
+                int tempTimeout = static_cast< int >( _changed_config.link.timeout );
+                if( ImGui::InputInt( "##Link Timeout (seconds)", &tempTimeout ) )
+                {
+                    _changed_config.link.timeout = static_cast< uint16_t >( std::max( 0, tempTimeout ) );
+                }
+            }
+            ImGui::RadioButton( "USB First", reinterpret_cast< int * >( &connection_priority ), 1 );
+            ImGui::RadioButton( "Dynamic Priority", reinterpret_cast< int * >( &connection_priority ), 2 );
+            switch( connection_priority )
+            {
+            case ETH_FIRST:
+                _changed_config.link.priority = link_priority::eth_first;
+                break;
+            case USB_FIRST:
+                _changed_config.link.priority = link_priority::usb_first;
+                break;
+            case DYNAMIC:
+                _changed_config.link.priority
+                    = _current_config.link.speed ? link_priority::dynamic_eth_first : link_priority::dynamic_usb_first;
```

Comment:
Logic not clear. Add a comment "If link speed is not 0 than we are connected by Ethernet"